backend/controllers/vehicleController.js

```diff
@@ -1,20 +1,31 @@
 const ErrorResponse = require('../utils/errorResponse');
 const asyncHandler = require('../middleware/async');
-const Vehicle = require('../models/Vehicle')
+const Vehicle = require('../models/Vehicle');
+const Dealership = require('../models/Dealership');
 
 // @desc        Get all vehicles for a specific dealership
 // @route       GET /api/v1/inventory/dealership/:dealershipId
 // @access      Private
 exports.getVehicles = asyncHandler(async (req, res, next) => {
+
+  // grab the dealership_ID from the body and verify that the dealership exists
+  const dealership = await Dealership.findById(req.params.dealershipId);
+
+  // no dealership found
+  if (!dealership) {
+      return next(
+          new ErrorResponse('dealership not found. Cannot return a list of vehicls without a valid dealership.', 404)
+      );
+  }
+  
   let query = Vehicle.find({ dealership: req.params.dealershipId });
   // run query
   const vehicles = await query;
-
-  if (!vehicles) {
-    return next(
-      new ErrorResponse(`No vehicles found with dealership id ${req.params.dealershipId}.`)
-    );
-  }
+  // if (!vehicles) {
+  //   return next(
+  //     new ErrorResponse(`No vehicles found with dealership id ${req.params.dealershipId}.`, 404)
+  //   );
+  // }
```

Comment:
We should probably still have this check in case the query doesn't return any vehicles, even though it should always return an array.

Reply:
The query always return an array. This array can be empty in two cases. If the provided dealership_id is invalid, or that dealership does not have any vehicles in it. I have added the check for the invalid dealership_id. Thus, this check is not necessary because it always returns an array

Reply:
Ok, in this case we should probably leave it out since it will decrease our code coverage percentage. NICE CATCH!